ONSdigital/takeon-business-layer
# Feature/spp 223 selective editing

# Description
This PR covers all the necessary work for the Selective Editing Business Layer logic. This includes the Graph QL query, API in getting Selective editing configuration(domain, current and previous responses, threshold, estimate, design weight etc) and API in saving Selective editing score and flag values. It is upsert save and also we are providing auditing information like last modified time and last modified user etc. This Business layer currently takes JSON payload of reference, period and survey to get the configuration data from database which then returns to the lambda. This Lambda performs calculations and outputs a score and a flag, which we then save to the database using another Business Layer endpoint - Save API.  
JIRA Link: https://collaborate2.ons.gov.uk/jira/browse/SPP-223
# Verify Code Coverage
From the terminal please run
mvn clean test
awk -F"," '{ instructions += $4 + $5; covered += $5 } END { print covered, "/", instructions, " instructions covered"; print 100*covered/instructions, "% covered" }' target/site/jacoco/jacoco.csv
Note: Compare code coverage in dev with the current story
# Verify Check style errors
From the terminal please run
mvn package install
# How to test?
In order to test, you should use the SPP-223 environment where the spp-es-takeon-validation-spp223-selective-editing Lambda sits.  
Run it manually in the console (no triggers yet, this is coming in a later story) using an example JSON payload:  
{
  "reference": "49900534932",
  "period": "201904",
  "survey": "023"
}
# Links
https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?spaceKey=SML&title=Example+mapping+-+Selective+Editing
https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?pageId=50795300
